### PR TITLE
Split build and test CI config files

### PR DESCRIPTION
**Pull request by shyamjesal**
Split go.yml into build and tests. taps and misc now execute on github runners.
Removed "TestCreateRemoveExtra" taps test.

Signed-off-by: Shyam Jesalpura <s.jesalpura@gmail.com>

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    branches: [ master ]
+  workflow_dispatch:
+
+env:
+  GOOS: linux
+  GO111MODULE: on
+
+jobs:
+  test:
+    name: Integration tests on Github Runners
```
Comment (ustiugov) [inline .github/workflows/tests.yml:16]: these are unit tests
rename: `Unit tests: $MODULE`.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    - name: Run tests in submodules
+      env:
+          MODULE: ${{ matrix.module }}
+      run: |
+        make -C $MODULE test
+        make -C $MODULE test-man
+  
+  ctriface-test:
+    name: ctriface & end-to-end integration tests on self hosted runners
```
Comment (ustiugov) [inline .github/workflows/tests.yml:42]: rename: `Unit tests: Firecracker-containerd interface`

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    - name: Run tests in submodules
+      run: |
+        make -C ctriface test
+        make -C ctriface test-man
+
+    - name: Cleaning
+      if: ${{ always() }}
+      run: ./scripts/clean_fcctr.sh
+
+    - name: Run vHive end-to-end tests
```
Comment (ustiugov) [inline .github/workflows/tests.yml:65]: rename: `Integration tests`. Split this into a separate job

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      run: make test
+
+    - name: Cleaning
+      if: ${{ always() }}
+      run: ./scripts/clean_fcctr.sh
+
+    - name: Run vHive tests that need manual cleaning
```
Comment (ustiugov) [inline .github/workflows/tests.yml:72]: rename: `vHive Integration tests (with manual cleaning)`. Split this into a separate job

File: taps/taps_test.go [hunk 5]
```diff
@@ -81,23 +81,23 @@ func TestCreateRemoveTaps(t *testing.T) {
 	}
 }
 
-func TestCreateRemoveExtra(t *testing.T) {
```
Comment (ustiugov) [inline taps/taps_test.go:84]: No, you need to use [Skip()](https://golang.org/pkg/testing/#hdr-Skipping). Also, add a comment that we disabled this test as it breaks in GitHub Actions and it doesn't seem essential for the test coverage

**Review by ustiugov — CHANGES_REQUESTED** [review]

**Comment by ustiugov** [comment]
I think that it would make sense to separate integration tests and unit tests into two files.

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      with:
+        go-version: 1.14
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: go build -race -v -a ./...
+
+    - name: Run tests in submodules
```
Comment (ustiugov) [inline .github/workflows/tests.yml:34]: Rename: Unit tests: ${{ matrix.module }}

File: .github/workflows/tests.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      env:
```
Comment (ustiugov) [inline .github/workflows/tests.yml:35]: remove these two lines, replace MODULE with ${{ matrix.module }}

**Comment by ustiugov** [comment]
Please replace the travis build badge with a single badge (or 3 badges: Build, Integration Tests, Unit Tests), as shown [here](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/adding-a-workflow-status-badge).

File: .github/workflows/tests.yml (hunk 1 above)
Comment (shyamjesal) [reply .github/workflows/tests.yml]: Done

File: .github/workflows/tests.yml (hunk 4 above)
Comment (shyamjesal) [reply .github/workflows/tests.yml]: Done. Separated integrations tests from unit tests

File: taps/taps_test.go (hunk 5 above)
Comment (shyamjesal) [reply taps/taps_test.go:84]: done

**Comment by ustiugov** [comment]
Let us skip this test (AllFunctions) on self-hosted-runners as well:
https://github.com/ease-lab/vhive/blob/8be91e903d100f4c8df7a7cb0592b6ee74ae84d6/fccd-orchestrator_test.go#L289

However, we need to run it in Travis (Travis runs only a nightly regression right now). Need to figure out how to do that though

File: .github/workflows/tests.yml (hunk 2 above)
Comment (shyamjesal) [reply .github/workflows/tests.yml]: Done

File: .github/workflows/tests.yml (hunk 3 above)
Comment (shyamjesal) [reply .github/workflows/tests.yml]: Done. Separated integrations tests

File: Makefile [hunk 8]
```diff
@@ -24,6 +24,7 @@ SUBDIRS:=ctriface taps misc
 EXTRAGOARGS:=-v -race -cover
 EXTRAGOARGS_NORACE:=-v
 EXTRATESTFILES:=fccd-orchestrator_test.go stats.go fccd-orchestrator.go functions.go
+SKIPTestAllFunctions:=-short
```
Comment (ustiugov) [inline Makefile:27]: let us just specify -short in case we will be excluding more tests later

File: Makefile [hunk 9]
```diff
@@ -44,16 +45,27 @@ test-orch: test test-man
...
 	sudo mkdir -m777 -p $(CTRDLOGDIR) && sudo env "PATH=$(PATH)" /usr/local/bin/firecracker-containerd --config /etc/firecracker-containerd/config.toml 1>$(CTRDLOGDIR)/fccd_orch_upf_log.out 2>$(CTRDLOGDIR)/fccd_orch_upf_log.err &
-	sudo env "PATH=$(PATH)" go test $(EXTRATESTFILES) $(EXTRAGOARGS) -args $(WITHSNAPSHOTS) $(WITHUPF)
+	sudo env "PATH=$(PATH)" go test $(EXTRATESTFILES) $(SKIPTestAllFunctions) $(EXTRAGOARGS) -args $(WITHSNAPSHOTS) $(WITHUPF)
 	./scripts/clean_fcctr.sh
 	sudo mkdir -m777 -p $(CTRDLOGDIR) && sudo env "PATH=$(PATH)" /usr/local/bin/firecracker-containerd --config /etc/firecracker-containerd/config.toml 1>$(CTRDLOGDIR)/fccd_orch_upf_lazy_log.out 2>$(CTRDLOGDIR)/fccd_orch_upf_lazy_log.err &
-	sudo env "PATH=$(PATH)" go test $(EXTRATESTFILES) $(EXTRAGOARGS) -args $(WITHSNAPSHOTS) $(WITHUPF) $(WITHLAZY)
+	sudo env "PATH=$(PATH)" go test $(EXTRATESTFILES) $(SKIPTestAllFunctions) $(EXTRAGOARGS) -args $(WITHSNAPSHOTS) $(WITHUPF) $(WITHLAZY)
 	./scripts/clean_fcctr.sh
 
+nightly-test:
```
Comment (ustiugov) [inline Makefile:58]: please move down right after `test-man-bench`

**Review by ustiugov — CHANGES_REQUESTED** [review]
cosmetic comments inlined.

File: Makefile (hunk 8 above)
Comment (shyamjesal) [reply Makefile]: done

File: Makefile (hunk 9 above)
Comment (shyamjesal) [reply Makefile]: Done

**Review by ustiugov — APPROVED** [review]

**Review by ustiugov — CHANGES_REQUESTED** [review]
oh, sorry. The badges are still missing

**Pull request merged**